interview-questions.md

```diff
@@ -7,49 +7,49 @@ First, without external resources. Challenge yourself to answer from memory as i
 
 3. What is a React component?
 
-Your answer:
+Your answer: React component is a folder under SRC to allow the App.js to look cleaner and use other functions on multiple components. The components have to be imported from App.js to invoke that compenent. Naming the name of your component under your header also passes the component from your App.js to your browser.
 
-Researched answer:
+Researched answer: React component consist of a type of function with JavaScript logic, these are called functional components. The goal of writing code is to have a the least ammount of code possible, by having components it allows the user to create a function that can be dynamic and prevent code from being repeated.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: State values are stored values that allow a variable to call on it later. An example would be const [myFood, setmyFood] = useState(0). Props are a way for components to pass data to each other, basically to simplify the code rather than hard coding.
+
 
-Researched answer:
+Researched answer: State is a special varaible strored to control the use of how that component will behave. Components that have stored varibales are considered a logical component. useState is a react hook that allows the user to update the value and implement a method to that value. Props allow the user to pass data and behavior between components, props are invoked very similar to how a function is invoked. The difference between state values and props are the control abilities from the component. State is owned by that particular component while props are controlled externally.
 
 5. What is the DOM?
 
-Your answer:
+Your answer: DOM is how a user interacts with the browser. A good example is the onClick button. The user clicks the button to interact with ther page. The DOM is the webpage that the user sees that allows for clicking or interaction.
 
-Researched answer:
+Researched answer: The DOM stands for Document Object Model, its what the user is looking at actual visual of the code that has been writen. DOM  is the representation of HTML on the displayed webpage. The actual website is the DOM, it allows interaction from the user to the page. 
 
 6. STRETCH: Which is the difference between a div tag and a span tag?
 
-Your answer:
+Your answer: Div tag basically creates a divider in the page, a block level element that will take up the whole distance from left to right. A user can either use a div tag or a an empty tag. A span tag would be to make the section bigger, by spanding out that certain section. 
 
-Researched answer:
+Researched answer:  Div and span tags are both considered containers however both have different methods. Span tags are used for grouping small text while div's are used for larger text. The div tag is used as a divider of the webpage while the span tag is used as a inline part of the page. 
```

Comment:
Exactly! Span will group small portions of text that we may want to style differently from the rest of the <p>.